Lib/importlib/abc.py

```diff
@@ -15,20 +15,36 @@
 import abc
 import warnings
 
-# for compatibility with Python 3.10
-from .resources.abc import ResourceReader, Traversable, TraversableResources
+from .resources import abc as _resources_abc
 
 
 __all__ = [
     'Loader', 'Finder', 'MetaPathFinder', 'PathEntryFinder',
     'ResourceLoader', 'InspectLoader', 'ExecutionLoader',
     'FileLoader', 'SourceLoader',
-
-    # for compatibility with Python 3.10
-    'ResourceReader', 'Traversable', 'TraversableResources',
 ]
 
 
+def __getattr__(name, canonical=_resources_abc):
+    """
+    For backwards compatibility, continue to make names
+    from canonical available through this module.
+    """
+    if name in canonical.__all__:
+        obj = getattr(canonical, name)
+        import warnings
+        warnings.warn(
```

Comment:
Instead of `warnings.warn`, let's use the new `warnings._deprecated`

https://discuss.python.org/t/introducing-warnings-deprecated/14856?u=hugovk

Reply:
Nice suggestion. TIL.

Added in 85259269b0.